scripts/patterns-reorg/models.py

```diff
@@ -111,12 +117,22 @@ def filter_entries(
 
 
 def determine_redirects(
-    entries: tuple[Entry, ...], *, prefix: str = ""
+    entries: tuple[Entry, ...] | tuple[DeletedPage, ...], *, prefix: str = ""
 ) -> dict[str, str]:
     result = {}
     for entry in entries:
-        if entry.slug and entry.from_file:
+        if entry.slug is not None and entry.from_file:
             old_url = str(PurePath(entry.from_file).with_suffix(""))
             result[old_url] = f"{prefix}{entry.slug.removeprefix('/')}"
-        result.update(determine_redirects(entry.children))
+
+            # We need to add two links for each index entry because we can
+            # have two links possible. For example, `/run/index` and `/run`
+            # point to the same page.
+            old_folder, old_file_name = old_url.split('/')
+            if old_file_name == "index":
+                result[f"{old_folder}/"] = f"{prefix}{entry.slug.removeprefix('/')}"
```

Comment:
You can have a new variable called `redirect_to` to DRY f"{prefix}{entry.slug.removeprefix('/')}"